experimental/Schemes/SpaceGerms.jl

```diff
@@ -0,0 +1,274 @@
+export SpaceGerm, germ_at_point
+
+export representative, point
+
+export ambient_germ
+
+import AbstractAlgebra: Ring
+
+@Markdown.doc """
+    AbsSpaceGerm{BaseRingType<:Ring,RingType<:Ring}
+
+A space germ, i.e. a ringed space ``(X,O_{(X,x)})`` with local ring ``O_{(X,x)}`` of type `RingType` over a ring ``k`` of type `BaseRingType`.
+"""
+abstract type AbsSpaceGerm{BaseRingType<:Ring, RingType<:Ring} <: AbsSpec{BaseRingType, RingType} end
+
+
+####################################################################
+## two short hand definitions for internal use only
+####################################################################
+GermAtClosedPoint = Spec{<:Field, 
+                         <:AbsLocalizedRing{<:Ring, <:Any, 
+                                            <:MPolyComplementOfKPointIdeal}
+                        }
+GermAtGeometricPoint = Spec{<:Field, 
+                            <:AbsLocalizedRing{<:Ring, <:Any, 
+                                               <:MPolyComplementOfPrimeIdeal}
+                           }
+
+###################################################################
+## start of the definition of space germ functionality
+###################################################################
+
+@Markdown.doc """
+    SpaceGerm{BaseRingType, RingType, SpecType}
+A space germ ``(X,O_{(X,x)}``, i.e. a ringed space with underlying scheme ``X`` of type SpecType and local ring ``O_{(X,x)}`` of type `RingType` over some base ring ``k`` of type `BaseRingType`.
+"""
+@attributes mutable struct SpaceGerm{BaseRingType<:Ring, RingType<:Ring, SpecType<:Spec} <: AbsSpaceGerm{BaseRingType, RingType}
+  X::SpecType
+
+  function SpaceGerm(X::GermAtClosedPoint)
+    return new{typeof(base_ring(X)), typeof(OO(X)), typeof(X)}(X)
+  end
+  
+  function SpaceGerm(X::GermAtGeometricPoint)
+    return new{typeof(base_ring(X)), typeof(OO(X)), typeof(X)}(X)
+  end
+end
+
+### Getter functions
+
+function underlying_scheme(X::SpaceGerm)
+  return X.X
+end
+
+@attr Spec function representative(X::SpaceGerm{<:Ring, <:MPolyQuoLocalizedRing})
+      return Spec(quotient_ring(OO(X)))
+end
+
+@attr Spec function representative(X::SpaceGerm{<:Ring, <:MPolyLocalizedRing})
+    R = ambient_ring(X)
+    return Spec(R)
+end
+
+function point(X::SpaceGerm{<:Any, <:Any, <:GermAtClosedPoint})
+  return point_coordinates(inverted_set(OO(X)))
```

Comment:
This does not return an ideal as was written above.

Reply:
corrected in documentation